### PR TITLE
Add active plugin list to compatibility service

### PR DESCRIPTION
**Pull request by allie500**
Fixes https://github.com/Automattic/woocommerce-payments-server/issues/3942

#### Changes proposed in this Pull Request

This PR adds the merchant site's active plugin list to the compatibility service.

#### Testing instructions

* Testing instructions are on https://github.com/Automattic/woocommerce-payments-server/pull/4369

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7933` or branch name `add/server-3942-compatibility-service-include-active-plugins` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/server-3942-compatibility-service-include-active-plugins) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: afdb9b7855f098cdc3c9e06e238213898941a3fe
- Build time: 2024-01-10 13:48:12 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: includes/class-compatibility-service.php [hunk 1]
```diff
@@ -47,11 +47,14 @@ public function init_hooks() {
 	 * @return void
 	 */
 	public function update_compatibility_data() {
+		$active_plugins = get_option( 'active_plugins' );
```
Comment (jessepearson) [inline includes/class-compatibility-service.php]: It should never happen, but `get_option` can return `false`. We should add a check and if `$active_plugins` is not an array, it should be changed to an empty array to match what's expected to be received at the server. 

A test would be needed to make sure that's working correctly, as well.

**Review by jessepearson — CHANGES_REQUESTED** [review]
Just the one (well, two because of the tests) change as a "just in case" approach.

File: includes/class-compatibility-service.php (hunk 1 above)
Comment (jessepearson) [reply includes/class-compatibility-service.php]: Resolving this per our discussion in slack: p1704818339032419-slack-C01B8KNUYSW

**Review by jessepearson — APPROVED** [review]
Approving pending updates on the server PR.

**Comment by jessepearson** [comment]
@allie500 I think a test like this can be added:

```php
	public function test_update_compatibility_data_active_plugins_false() {
		// Arrange: Create the expected value to be passed to update_compatibility_data.
		$expected = [
			'woopayments_version' => WCPAY_VERSION_NUMBER,
			'woocommerce_version' => WC_VERSION,
			'active_plugins'      => [],
		];

		$this->break_active_plugins_option();

		// Arrange/Assert: Set the expectations for update_compatibility_data.
		$this->mock_api_client
			->expects( $this->once() )
			->method( 'update_compatibility_data' )
			->with( $expected );

		// Act: Call the method we're testing.
		$this->compatibility_service->update_compatibility_data();

		$this->fix_active_plugins_option();
	}

	private function break_active_plugins_option() {
		update_option( 'temp_active_plugins', get_option( 'active_plugins' ) );
		delete_option( 'active_plugins' );
	}

	private function fix_active_plugins_option() {
		update_option( 'active_plugins', get_option( 'temp_active_plugins' ) );
		delete_option( 'temp_active_plugins' );
	}
```

**Comment by allie500** [comment]
> @allie500 I think a test like this can be added:
> 
> ```
> 	public function test_update_compatibility_data_active_plugins_false() {
> 		// Arrange: Create the expected value to be passed to update_compatibility_data.
> 		$expected = [
> 			'woopayments_version' => WCPAY_VERSION_NUMBER,
> 			'woocommerce_version' => WC_VERSION,
> 			'active_plugins'      => [],
> 		];
> 
> 		$this->break_active_plugins_option();
> 
> 		// Arrange/Assert: Set the expectations for update_compatibility_data.
> 		$this->mock_api_client
> 			->expects( $this->once() )
> 			->method( 'update_compatibility_data' )
> 			->with( $expected );
> 
> 		// Act: Call the method we're testing.
> 		$this->compatibility_service->update_compatibility_data();
> 
> 		$this->fix_active_plugins_option();
> 	}
> 
> 	private function break_active_plugins_option() {
> 		update_option( 'temp_active_plugins', get_option( 'active_plugins' ) );
> 		delete_option( 'active_plugins' );
> 	}
> 
> 	private function fix_active_plugins_option() {
> 		update_option( 'active_plugins', get_option( 'temp_active_plugins' ) );
> 		delete_option( 'temp_active_plugins' );
> 	}
> ```

Thanks for the suggestion. I've added the test in https://github.com/Automattic/woocommerce-payments/pull/7933/commits/afdb9b7855f098cdc3c9e06e238213898941a3fe.

**Review by jessepearson — APPROVED** [review]
Everything is working as expected. Thanks for all of the updates done here 😁

**Pull request merged**